cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -41,6 +41,11 @@ func runProfile(cmd *cobra.Command, args []string) {
 	validateArgs(args)
 	profileName := args[0]
 
+	err := removeOcContext(profileActions.GetActiveProfile())
```

Comment:
I am not sure we should do this regardless. Have you actually tries then switching back to the original profile. It seems deleting the current context and then returning to the profile, leaves me logged out. 

    $ minishift profile set snafu
    No oc CLI context could be activated for the 'snafu' profile.
    Profile 'snafu' set as active profile
    $ minishift profile set minishift
    Profile 'minishift' set as active profile
    $  oc get bc
    User "system:anonymous" cannot list buildconfigs in project "myproject"

I was logged in as admin before. Some information is lost here.

Reply:
btw, you are unsetting the current context not removing it. The function name seems misleading.

Reply:
@hferentschik I couldn't reproduce this, i get an error saying `No resources found`